nmosauth/auth_server/security_api.py

```diff
@@ -286,15 +316,19 @@ def delete_user(self, username):
         removeResourceOwner(username)
         return redirect(url_for('_get_users'))
 
-    # route for certificate with public key
-    @route(AUTH_VERSION_ROOT + 'certs/', methods=['GET'], auto_json=True)
-    def get_cert(self):
+    # route for JSON Web Key
+    @route(AUTH_VERSION_ROOT + JWK_ENDPOINT + '/', methods=['GET'], auto_json=True)
+    def get_jwk(self):
         try:
-            with open(CERT_PATH, 'r') as myfile:
-                cert = myfile.read()
-            return (200, [cert])
+            with open(PUBKEY_PATH, 'r') as myfile:
+                pub_key = myfile.read()
+            obj = jwk.dumps(
+                pub_key, kty='RSA', use="sig", key_ops="verify",
+                alg="RS512", kid=date.today().strftime("%d-%m-%y")
+            )
+            return (200, obj)
         except OSError as e:
-            self.logger.writeError("Error: {}\nFile at {} doesn't exist".format(e, CERT_PATH))
+            self.logger.writeError("Error: {}\nFile at {} doesn't exist".format(e, PUBKEY_PATH))
```

Comment:
Does this want to raise an exception or should it just return an error to the client?

Reply:
It should probably fail, as without the presence of the JWK, there would be no way for client's to validate tokens. Prevents the server failing silently.